packages/apps/shopify-app-remix/src/server/clients/types.ts

```diff
@@ -29,10 +30,15 @@ export interface GraphQLQueryOptions<
   tries?: number;
 }
 
+export interface FetchResponse<TData = any>
+  extends Omit<FetchResponseBody<TData>, 'headers'> {
+  headers?: Headers;
+}
+
 export type GraphQLResponse<
   Operation extends keyof Operations,
   Operations extends AllOperations,
-> = ResponseWithType<FetchResponseBody<ReturnData<Operation, Operations>>>;
+> = ResponseWithType<FetchResponse<ReturnData<Operation, Operations>>>;
```

Comment:
Will this return `headers` as an object or as a `Headers` type? In the remix package, we're returning a proper `Response` object, so the headers should be the fetch API ones too.